### PR TITLE
fix sequence layout placements

### PR DESCRIPTION
**Pull request by alixander**
Screen Shot 2022-12-05 at 1 56 02 PM

### CODE REVIEW

File: d2exporter/export.go [hunk 1]
```diff
@@ -99,6 +100,10 @@ func toShape(obj *d2graph.Object, theme *d2themes.Theme) d2target.Shape {
 	shape.Italic = text.IsItalic
 	shape.FontSize = text.FontSize
 
+	if strings.EqualFold(obj.Attributes.Shape.Value, d2target.ShapeSequenceDiagram) {
```
Comment (ejulio-ts) [inline d2exporter/export.go:103]: obj.IsSequenceDiagram

File: d2graph/d2graph.go [hunk 2]
```diff
@@ -339,8 +327,26 @@ func (obj *Object) GetFill(theme *d2themes.Theme) string {
...
+			return theme.Colors.B6
+		} else if level == 3 {
+			return theme.Colors.Neutrals.N6
+		}
+		return theme.Colors.Neutrals.N7
+	}
+
 	shape := obj.Attributes.Shape.Value
 
+	if strings.EqualFold(shape, d2target.ShapeSequenceDiagram) {
```
Comment (ejulio-ts) [inline d2graph/d2graph.go:346]: same as above

File: d2layouts/d2sequence/layout.go [hunk 3]
```diff
@@ -141,14 +141,26 @@ func cleanup(g *d2graph.Graph, sequenceDiagrams map[string]*sequenceDiagram, obj
...
+		)
 
 		obj.Children = make(map[string]*d2graph.Object)
+		obj.ChildrenArray = make([]*d2graph.Object, 0)
 		for _, child := range sd.actors {
 			obj.Children[child.ID] = child
+			obj.ChildrenArray = append(obj.ChildrenArray, child)
+		}
+		for _, child := range sd.groups {
+			if child.Parent == obj {
```
Comment (ejulio-ts) [inline d2layouts/d2sequence/layout.go:159]: something tells me this won't work with TALA as the object references change in that call

Comment (alixander) [reply d2layouts/d2sequence/layout.go]: changed to IDs. that'll work right

**Pull request merged**